### PR TITLE
mgr/dashboard: paginate services

### PR DESCRIPTION
**Pull request by pereman2**
Signed-off-by: Pere Diaz Bou <pdiazbou@redhat.com>
Fixes: https://tracker.ceph.com/issues/56512

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts [hunk 1]
```diff
@@ -385,96 +388,100 @@ export class ServiceFormComponent extends CdForm implements OnInit {
...
-            const specKeys = ['pool', 'api_password', 'api_user', 'trusted_ip_list', 'api_port'];
-            specKeys.forEach((key) => {
-              this.serviceForm.get(key).setValue(response[0].spec[key]);
-            });
-            this.serviceForm.get('ssl').setValue(response[0].spec?.api_secure);
-            if (response[0].spec?.api_secure) {
-              this.serviceForm.get('ssl_cert').setValue(response[0].spec?.ssl_cert);
-              this.serviceForm.get('ssl_key').setValue(response[0].spec?.ssl_key);
+      this.cephServiceService
+        .list(new HttpParams({ fromObject: { limit: -1, offset: 0 } }), this.serviceName)
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts:392]: @pereman2 do we actually pagination when we call the list endpoint with a known service name?

**Review by nizamial09 — COMMENTED** [review]
@pereman2 only issue I stumbled upon is this one. If you try to edit a service, instead of showing the details of the service I selected to edit, it always shows the details ofalertmanager (or maybe the first one in the list) in the form
.
https://user-images.githubusercontent.com/71764184/191276360-d13ba956-9629-4eb6-ac57-dfdf06e637c0.mp4

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts (hunk 1 above)
Comment (pereman2) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts:392]: I wasn't properly updating the `httpParams` argument in `cephServiceService.list`. This should be fixed by now, please re review.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/models/cd-table-fetch-data-context.ts [hunk 2]
```diff
@@ -23,15 +23,22 @@ export class CdTableFetchDataContext {
   }
 
   toParams(): HttpParams {
+    if (this.pageInfo.offset == NaN) {
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/models/cd-table-fetch-data-context.ts:26]: I would prefer [isNan](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#using_isnan)
```suggestion
    if (Number.isNaN(this.pageInfo.offset)) {
```

**Review by nizamial09 — APPROVED** [review]
Tested locally and looks good to me!

File: src/pybind/mgr/dashboard/frontend/src/app/shared/models/cd-table-fetch-data-context.ts (hunk 2 above)
Comment (pereman2) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/models/cd-table-fetch-data-context.ts]: done!

**Comment by pereman2** [comment]
jenkins test dashboard

**Review by Pegonzal — APPROVED** [review]
Tested it locally, all good!

**Pull request merged**